From a82e9304ecfe2a4c5be501c49a7d0c782ba64d34 Mon Sep 17 00:00:00 2001
From: "Ooi, Joyce" <joyce.ooi@intel.com>
Date: Thu, 19 Mar 2020 21:49:50 +0800
Subject: [PATCH 029/120] HSD #1507870299: video: fbdev: fix dma allocation
 error

commit 1f98f60c7bbe51e9fbeeb019791221a1b5b31c68 from
https://github.com/altera-opensource/linux-socfpga.git

A NULL pointer is being passed into the dma_alloc_coherent as a 'dev'
pointer, which causes a kernel panic as it tries to dereference a NULL
pointer. This patch fixes the problem by passing a device into the
function. The same is applied for dma_free_coherent.

Signed-off-by: Ooi, Joyce <joyce.ooi@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/video/fbdev/altvipfb2.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/video/fbdev/altvipfb2.c b/drivers/video/fbdev/altvipfb2.c
index ef95a88ce2ea..7e6af2302227 100644
--- a/drivers/video/fbdev/altvipfb2.c
+++ b/drivers/video/fbdev/altvipfb2.c
@@ -133,7 +133,7 @@ int altvipfb2_probe(struct device *dev, void __iomem *base)
 	void *fbmem_virt;
 	struct altvipfb2_priv *fbpriv = dev_get_drvdata(dev);
 
-	fbmem_virt = dma_alloc_coherent(NULL,
+	fbmem_virt = dma_alloc_coherent(dev,
 					fbpriv->info.fix.smem_len,
 					(void *)&fbpriv->info.fix.smem_start,
 					GFP_KERNEL);
@@ -163,7 +163,7 @@ int altvipfb2_probe(struct device *dev, void __iomem *base)
 
 err_dma_free:
 	fb_dealloc_cmap(&fbpriv->info.cmap);
-	dma_free_coherent(NULL, fbpriv->info.fix.smem_len, fbmem_virt,
+	dma_free_coherent(dev, fbpriv->info.fix.smem_len, fbmem_virt,
 			  fbpriv->info.fix.smem_start);
 	return retval;
 }
@@ -174,7 +174,7 @@ int altvipfb2_remove(struct device *dev)
 	struct altvipfb2_priv *fbpriv = dev_get_drvdata(dev);
 
 	altvipfb2_disable_hw(fbpriv->base);
-	dma_free_coherent(NULL, fbpriv->info.fix.smem_len,
+	dma_free_coherent(dev, fbpriv->info.fix.smem_len,
 			(void *)&fbpriv->info.screen_base,
 			fbpriv->info.fix.smem_start);
 
-- 
2.31.1

